src/java.desktop/share/classes/java/awt/Graphics2D.java

```diff
@@ -1202,8 +1202,8 @@ public abstract boolean hit(Rectangle rect,
      * argument, the specified {@code Shape} becomes the new
      * user clip.
      * @param s the {@code Shape} to be intersected with the current
-     *          {@code Clip}.  If {@code s} is {@code null},
-     *          this method clears the current {@code Clip}.
+     *          {@code Clip}.
+     * @throws NullPointerException if {@code s} is {@code null}
```

Comment:
Actually it is not always true, you can check it by commenting `setClip()` call in the test.

Reply:
The spec says "s - the Shape to be intersected with the current Clip" so I assume it means there should be a current clip set, so that is why I have used setClip to "set" a clip. So, setClip() should be there as far I see.

Reply:
Also, if there is no clip set, then the spec statement " If s is null, this method clears the current Clip" does not carry any meaning, so in both regard, setClip() should be there, I presume.

Reply:
@prsadhuk At first glance i also thought clip() should be called after calling setClip() but that is not the case.

If we see SunGraphics2D implementation of clip() we dont exit if there is no clip(usrClip object) is set using setClip(). So clip() doesnt depend on whether setClip() is used or not.

Reply:
> Also, if there is no clip set, then the spec statement " If s is null, this method clears the current Clip" does not carry any meaning, so in both regard, setClip() should be there, I presume.

The old javadoc is definitely does not conform the current behavior. But as of now it clearly says that it will throw NPE if null argument passed.

Reply:
Looks like this is just a bug in the implementation, the null should reset the clip.

Reply:
I am not sure if it's implementation bug. As Jim Graham has mentioned in thebug, 

> clip(null) is not legal.
> 
> It is *setClip*(null) that clears the clip.
I could rephrase the doc to specify NPE willbe thrrown for null Shape if a clip is already set.

Reply:
If "*setClip*(null)" has to clear the clip then it should be specified, currently, that method said nothing about the null parameter.

Reply:
As per code 
```
public void setClip(Shape sh) {
        usrClip = transformShape(sh);
```

usrClip is set to null if "sh" is null so clip is cleared...I will update the setClip doc too..

Reply:
From what I can tell, if clip is null, then calling clip(null) has no effect.
And it won't throw an NPE in that case. So the proposed documentation is less accurate than no documentation at all.
And I would shy away from changing the implementation on this without LOTS of VERY careful testing.

Reply:
> usrClip is set to null if "sh" is null so clip is cleared...I will update the setClip doc too..

Huh ? Not understanding you. If the userClip is non-null then we enter this
        if (usrClip != null) {
            s = intersectShapes(usrClip, s, true, true);
        }
which i where you get the NPE from you are proposing to document.

Reply:
It seems to me to be a little accidental from the implementation that clip(null) will just silently return
if the user clip is currently null.
But I think it unduly risky to "clean that up" after so many years.
So I think we are stuck with some variant on the weird wording.
The bit about using setClip is bogus. I suggest adding the clause worded like this
@throws NullPointerException if {@code s} is {@ code null) and a user clip is currently set.

and explanatory text like this
Since this method intersects the specified shape with the current clip, it will throw NPE for a null shape
unless the user clip is also null. So calling this method with a null argument is not recommended.